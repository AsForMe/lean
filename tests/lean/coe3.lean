constants A B₁ B₂ C D : Type

constant A_to_B₁ : has_coe A B₁
constant A_to_B₂ : has_coe A B₂
constant B₁_to_C : has_coe B₁ C
constant B₂_to_D : has_coe B₂ D

attribute [instance] A_to_B₁ A_to_B₂ B₁_to_C B₂_to_D

constant a : A
constant f : C → C
constant g : D → D

#check f (a : B₁)

#check g (a : B₂)
