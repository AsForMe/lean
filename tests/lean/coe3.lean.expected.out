f ↑↑a : C
g ↑↑a : D
